vyked/registry.py

```diff
@@ -128,17 +138,324 @@ def _get_full_service_name(service: str, version):
+        #     for each in vendors:
+        #         if each['service'] == service_name and each['version'] == service_version:
+        #             consumers.add(self._split_key(service))
+        #
+        # print(consumers)
+        # return consumers
+
+    def get_vendors(self, service, version):
+        #TODO: select child.* from dependencies where service, version
+        rows = self.qselect('dependencies', 'parent_name',columns=['parent_name', 'parent_version'],
+                                      where_keys=[{'child_name': ('=', service),
+                                                   'child_version': ('=', version)}])
+        print("get_vendors")
+        crows = [{'service': s, 'version': v} for (s,v) in rows]
+        return crows
+        # print(crows)
+        # print(self._service_dependencies.get(self._get_full_service_name(service, version), []))
+        # return self._service_dependencies.get(self._get_full_service_name(service, version), [])
+
+    def get_node(self, node_id):
+        #TODO: select * from services where node_id
+        rows = self.qselect('services', 'service_name',
+                            columns=['service_name','version', 'ip', 'port', 'id', 'protocol'],
+                            where_keys=[{'id': ('=', node_id)}])
+        print(rows)
+        for name, version, host, port, node, protocol in rows:
+            return Service(name, version, [], host, port, node, protocol)
+        # for name, versions in self._registered_services.items():
+        #     for version, instances in versions.items():
+        #         for host, port, node, service_type in instances:
+        #             if node_id == node:
+        #                 return Service(name, version, [], host, port, node, service_type)
+        return None
+
+    def remove_node(self, node_id):
+        #TODO: delete from services where node_id
+        self.qdelete('services', where_keys=[{'id': ('=', node_id)}])
+        # for name, versions in self._registered_services.items():
+        #     for version, instances in versions.items():
+        #         for instance in instances:
+        #             host, port, node, service_type = instance
+        #             if node_id == node:
+        #                 instances.remove(instance)
+        #TODO: insert into uptimes (node_id, downtime, time.time())
+        uptimes_dict = {'id': node_id, 'event_type': 'DOWNTIME', 'event_time': int(time.time())}
+        self.qinsert('uptimes', uptimes_dict)
+        # for name, nodes in self._uptimes.items():
+        #     for node, uptimes in nodes.items():
+        #         if node == node_id:
+        #             uptimes['downtime'] = int(time.time())
+        return None
+
+    def get_uptimes(self):
+        #TODO: select * from uptimes
+        rows = self.qselect('uptimes', 'id', columns=['id', 'event_type', 'event_time'])
+        print("get_uptimes")
```

Comment:
@amanwriter get rid of print statements. Wherever necessary, write out to logs instead.